src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.HotReload.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Threading.Tasks;
+
+[assembly: ElementMetadataUpdateHandlerAttribute(typeof(Windows.UI.Xaml.Controls.ScrollViewer), typeof(Windows.UI.Xaml.Controls.ScrollViewerMetadataUpdateHandler))]
+
+namespace Windows.UI.Xaml.Controls;
+
+internal static partial class ScrollViewerMetadataUpdateHandler
+{
+	public static void CaptureState(FrameworkElement element, IDictionary<string, object> stateDictionary, Type[] updatedTypes)
+	{
+		stateDictionary["VOffset"] = (element as ScrollViewer)?.VerticalOffset ?? 0.0;
+		stateDictionary["HOffset"] = (element as ScrollViewer)?.HorizontalOffset ?? 0.0;
+	}
+
+	public static async void RestoreState(FrameworkElement element, IDictionary<string, object> stateDictionary, Type[] updatedTypes)
+	{
+		if (element is ScrollViewer sv)
+		{
+			// TODO: Is there a better way to ensure the scrollviewer has realized its children than to do a yield?
+			// And should we make the Capture/Restore async to support yielding?
+			await Task.Yield();
```

Comment:
cc @jeromelaban is there a better way to make sure the UI has updated prior to setting the scroll offset. The issue is that if we attempt to set scroll offset immediately, the ScrollableHeight is still 0 as the children elements haven't had a chance to load.
Also, should we make Restore async so that we can yield here. This would make applying updates more determininistic, but would mean that the ui update phase will take longer to complete

Reply:
There isn't an explicit way to do so, no, aside from waiting for the `Loaded` event. If we dice to make this async, let's return proper tasks and avoid async void :)

Reply:
Changed restore to be async Task